nav2_bringup/bringup/params/nav2_params_voxel.yaml

```diff
@@ -0,0 +1,192 @@
+amcl:
+  ros__parameters:
+    use_sim_time: True
+    alpha1: 0.2
+    alpha2: 0.2
+    alpha3: 0.2
+    alpha4: 0.2
+    alpha5: 0.2
+    base_frame_id: "base_footprint"
+    beam_skip_distance: 0.5
+    beam_skip_error_threshold: 0.9
+    beam_skip_threshold: 0.3
+    do_beamskip: false
+    global_frame_id: "map"
+    lambda_short: 0.1
+    laser_likelihood_max_dist: 2.0
+    laser_max_range: 100.0
+    laser_min_range: -1.0
+    laser_model_type: "likelihood_field"
+    max_beams: 60
+    max_particles: 2000
+    min_particles: 500
+    odom_frame_id: "odom"
+    pf_err: 0.05
+    pf_z: 0.99
+    recovery_alpha_fast: 0.0
+    recovery_alpha_slow: 0.0
+    resample_interval: 1
+    robot_model_type: "differential"
+    save_pose_rate: 0.5
+    sigma_hit: 0.2
+    tf_broadcast: true
+    transform_tolerance: 1.0
+    update_min_a: 0.2
+    update_min_d: 0.25
+    z_hit: 0.5
+    z_max: 0.05
+    z_rand: 0.5
+    z_short: 0.05
+
+amcl_map_client:
+  ros__parameters:
+    use_sim_time: True
+
+amcl_rclcpp_node:
+  ros__parameters:
+    use_sim_time: True
+
+bt_navigator:
+  ros__parameters:
+    use_sim_time: True
+    bt_xml_filename: "navigate_w_replanning_and_recovery.xml"
+
+bt_navigator_rclcpp_node:
+  ros__parameters:
+    use_sim_time: True
+
+controller_server:
+  ros__parameters:
+    use_sim_time: True
+    debug_trajectory_details: True
+    min_vel_x: 0.0
+    min_vel_y: 0.0
+    max_vel_x: 0.26
+    max_vel_y: 0.0
+    max_vel_theta: 1.0
+    min_speed_xy: 0.0
+    max_speed_xy: 0.26
+    min_speed_theta: 0.0
+    min_x_velocity_threshold: 0.001
+    # Add high threshold velocity for turtlebot 3 issue.
+    # https://github.com/ROBOTIS-GIT/turtlebot3_simulations/issues/75
+    min_y_velocity_threshold: 0.5
+    min_theta_velocity_threshold: 0.001
+    acc_lim_x: 2.5
+    acc_lim_y: 0.0
+    acc_lim_theta: 3.2
+    decel_lim_x: -2.5
+    decel_lim_y: 0.0
+    decel_lim_theta: -3.2
+    vx_samples: 20
+    vy_samples: 5
+    vtheta_samples: 20
+    sim_time: 1.7
+    linear_granularity: 0.05
+    xy_goal_tolerance: 0.25
+    transform_tolerance: 0.2
+    critics: ["RotateToGoal", "Oscillation", "BaseObstacle", "GoalAlign", "PathAlign", "PathDist", "GoalDist"]
+    BaseObstacle.scale: 0.02
+    PathAlign.scale: 0.0
+    GoalAlign.scale: 0.0
+    PathDist.scale: 32.0
+    GoalDist.scale: 24.0
+    RotateToGoal.scale: 32.0
+
+controller_server_rclcpp_node:
+  ros__parameters:
+    use_sim_time: True
+
+local_costmap:
+  local_costmap:
+    ros__parameters:
+      use_sim_time: True
+      global_frame: odom
+      plugin_names: ["voxel_layer", "inflation_layer"]
+      plugin_types: ["nav2_costmap_2d::VoxelLayer", "nav2_costmap_2d::InflationLayer"]
+      clearable_layers: ["voxel_layer"]
+      rolling_window: true
+      width: 3
+      height: 3
+      resolution: 0.05
+      robot_radius: 0.22
+      inflation_layer.cost_scaling_factor: 3.0
+      obstacle_layer:
```

Comment:
There is no obstacle layer defined in this configuration. Remove please

Reply:
voxel_layer also initializes obstacle_layer, so I believe obstacle_layer params are required. When I tried without specifying obstacle layer params, obstacles would not show up in the global costmap.

Reply:
Thats a bug that needs to be fixed. 

Voxel layer _derives_ from the obstacle layer, but the base class should *not* have its parameters namespaced in the bass class. 

In this topic, one could make an argument about software practice, but the stronger argument is that a user never defined an obstacle layer, why must they add parameters for one? That’s confusing